### ISSUE TITLE
Suppress Checker Framework warnings in AutoValue generated classes.

### ISSUE THREAD

**Issue by sebright** [census-instrumentation/opencensus-java#917]
~~https://github.com/census-instrumentation/opencensus-java/pull/916/commits/18cf999ddea65775db2a482e00cea05d52961077~~ bbe6b10f7c6b622b4ffd18e98f58d96a8eba3854 used `SuppressWarnings("nullness")` on all AutoValue abstract classes to suppress warnings in the generated subclasses.  The warnings were caused by missing Nullable annotations on the parameter to `equals`.  It would be better to find a way to only suppress warnings in the generated classes.  Another solution would be to make AutoValue add the correct annotations.

**Comment by sebright** [comment]
I opened an issue for adding the correct annotations with AutoValue: https://github.com/google/auto/issues/579

**Comment by sebright** [comment]
The AutoValue issue has been addressed, so we can remove the warning suppression after the feature is released and we upgrade AutoValue.  Upgrading AutoValue depends on #1017.

**Comment by sebright** [comment]
AutoValue 1.5.4 contains the fix: https://github.com/google/auto/releases/tag/auto-value-1.5.4

**Comment by sebright** [comment]
We probably need to upgrade to 1.6 when it is released, so that we can get a bug fix: https://github.com/google/auto/commit/6607dd749a8c345e9f6787d55d1e484d40588676

**Comment by sebright** [comment]
I tried upgrading to AutoValue 1.6 and adding a Nullable anotation to `equals`, but then I got a lot of other warnings.  For example, the Checker Framework complained that nullable fields weren't annotated with `@Nullable` and AutoValue builders didn't initialize all fields in the constructor.  Then I found that I could suppress warnings in all generated AutoValue classes with the flag `-AskipDefs` (https://checkerframework.org/manual/#askipdefs).
